didcomm_messaging/quickstart.py

```diff
@@ -0,0 +1,393 @@
+from didcomm_messaging.resolver.peer import Peer2, Peer4
+from didcomm_messaging.routing import RoutingService
+
+JSON_OBJ = Dict[str, Any]
+Attachment = JSON_OBJ
+JSON_VALUE = Union[None, str, int, bool, float, JSON_OBJ, List[Any]]
+
+LOG = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class Message:
+    """Provide a nicer interface for messages than just Dictionaries."""
+
+    type: str
+    body: JSON_OBJ
+    id: Optional[str] = None
+    typ: Optional[str] = "application/didcomm-plain+json"
+    frm: Optional[DID] = None
+    to: Optional[List[DID]] = None
+    lang: Optional[str] = None
+    created_time: Optional[int] = None
+    expires_time: Optional[int] = None
+    thid: Optional[str] = None
+    pthid: Optional[str] = None
+    please_ack: Optional[List[str]] = None
+    ack: Optional[List[str]] = None
+
+    # TODO: better implement/support these fields
+    attachments: Optional[List[Attachment]] = None
+    # from_prior: Optional[JWT] = None
+
+    # TODO: Add message validation for spec-defined fields
+
+    def __attrs_post_init__(self):
+        """Ensure that id is set."""
+        # super().__init__(*args, **kwargs)
+        if self.id is None:
+            self.id = str(uuid.uuid4())
+
+    def as_dict(self):
+        """Return message as a dictionary for serialization."""
+        return attrs.asdict(self, filter=(lambda _, x: x is not None))
+
+    @classmethod
+    def from_json(cls, data: str):
+        """Create a Message object from a JSON string."""
+        data = json.loads(data)
+        if "from" in data:
+            data["frm"] = data["from"]
+            del data["from"]
+        return cls(**data)
+
+
+class CompatibilityPrefixResolver(PrefixResolver):
+    """Provide backwards compatibility with older DID methods.
+
+    This will be removed in the future, as the intent is for agents to follow
+    the did:peer:2 spec. The CompatibilityPrefixResolver allows for interaction
+    with agents that are using the old #key-byte-prefix. Once this changes,
+    this class will be removed.
+
+    If you don't need this, it is recommended that you just use the
+    PrefixResolver directly.
+    """
+
+    async def resolve_and_parse(self, did: str) -> DIDDocument:
+        """Resolve a DID and parse the DID document."""
+        doc = await self.resolve(did)
+        # return DIDDocument.deserialize(doc)
+        id_map = {}
+
+        def set_id(method):
+            new_id = method["publicKeyMultibase"][1:9]
+            id_map[method["id"]] = new_id
+            method["id"] = did + "#" + new_id
+            return method
+
+        doc["verificationMethod"] = [
+            set_id(method) for method in doc["verificationMethod"]
+        ]
+        doc["authentication"] = [
+            did + "#" + id_map.get(id) for id in doc["authentication"]
+        ]
+        doc["keyAgreement"] = [did + "#" + id_map.get(id) for id in doc["keyAgreement"]]
+        return DIDDocument.deserialize(doc)
```

Comment:
Thoughts on working to get this removed before we merge this PR? I want to bury the old did:peer:2 semantics as soon as possible lol.

Reply:
Plus, it would simplify this quickstart even more.

Reply:
It's partially in here as that's what is required to talk to the DIDComm demo at present. As soon as both of those have moved away from using the only method, I intend to drop the compatibility pieces